### PR TITLE
Fixes #25493 - simplify Setting cache handling

### PR DESCRIPTION
**Pull request by tbrisker**

### CODE REVIEW

**Review by ohadlevy — DISMISSED** [review]
I was trying really hard to remember why we didnt use fetch to begin with, and couldn't.... ACK pending tests.

**Comment by ohadlevy** [comment]
@tbrisker test failures seems relevant.

File: app/models/setting.rb [hunk 1]
```diff
@@ -92,13 +92,8 @@ def self.humanized_category
...
   def self.[](name)
     name = name.to_s
-    cache_value = Setting.cache.read(name)
-    if cache_value.nil?
-      value = find_by(:name => name).try(:value)
-      Setting.cache.write(name, value)
-      return value
-    else
-      return cache_value
+    cache.fetch("settings/#{name}") do
```
Comment (ekohl) [inline app/models/setting.rb]: Would it be a good practice to put the cache key in a function? Now we're repeating magic strings in our code. At least I'd expect a `CACHE_KEY_PREFIX` constant

Comment (tbrisker) [reply app/models/setting.rb]: added a method, though tbh the tests seem to test that rails caching works...

File: app/models/setting.rb [hunk 2]
```diff
@@ -357,11 +355,15 @@ def clear_value_when_default
 
   def clear_cache
     # ensures we don't have cache left overs in settings
-    if Setting.cache.delete(name.to_s) == false
+    unless Setting.cache.delete(cache_key)
```
Comment (ekohl) [inline app/models/setting.rb:358]: This should be `cache_key(name)` though I'm not sure where name comes from here

Comment (ekohl) [reply app/models/setting.rb:358]: Never mind, this is the instance vs class method

**Review by ekohl — APPROVED** [review]

**Comment by tbrisker** [comment]
Test failures are unrelated

**Review by lzap — APPROVED** [review]

**Comment by lzap** [comment]
Thanks merged.

**Comment by leewaa** [comment]
This is causing me a ton of warnings (108 to be exact), even with a fresh database: 
```
$ bundle exec rails db:drop
. . .
$ bundle exec rails db:create
. . .
$ bundle exec rails db:migrate
. . .
$ bundle exec rails db:seed
```
```
. . . 
2018-11-20T16:34:05 [W|app|] Failed to remove administrator from cache
2018-11-20T16:34:05 [W|app|] Failed to remove foreman_url from cache
2018-11-20T16:34:05 [W|app|] Failed to remove entries_per_page from cache
2018-11-20T16:34:05 [W|app|] Failed to remove fix_db_cache from cache
2018-11-20T16:34:05 [W|app|] Failed to remove max_trend from cache
2018-11-20T16:34:05 [W|app|] Failed to remove db_pending_migration from cache
2018-11-20T16:34:05 [W|app|] Failed to remove db_pending_seed from cache
2018-11-20T16:34:05 [W|app|] Failed to remove proxy_request_timeout from cache
2018-11-20T16:34:05 [W|app|] Failed to remove login_text from cache
. . . 
```

**Comment by tbrisker** [comment]
hmmm... look like the rails documentation for cache.delete isn't clear, it returns nil if the key is not found and false if there is an error. Follow up PR in a moment.

**Comment by tbrisker** [comment]
@leewaa thanks for noticing. GH-6267 should fix this.

**Comment by leewaa** [comment]
@tbrisker No problem :) will take a look

**Pull request merged**